### PR TITLE
[Do not merge][Testing][ETL] Introducing a new API to extract samples for ETL operations

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Fresh API for retrieving a sample of 100 rows.

## How was this patch tested?

- Pending

### CODE REVIEW

**Comment by agl29** [comment]
@sreenaths can you please check this change with your UI changes? And let me know if more changes are needed.

File: desktop/libs/indexer/src/indexer/api3.py [hunk 1]
```diff
@@ -259,6 +259,39 @@ def decode_utf8(input_iterator):
   for l in input_iterator:
     yield l.decode('utf-8')
 
+
+def etl_sample_rows(request):
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/api3.py:263]: Suggestion: To add the error handling decorator.

File: desktop/libs/indexer/src/indexer/api3.py [hunk 2]
```diff
@@ -259,6 +259,39 @@ def decode_utf8(input_iterator):
...
+  for count, csv_row in enumerate(reader):
+    if format['hasHeader'] and count == 0:
+      continue
+    sample.append(csv_row)
+    if count == 99:
+      break
+
+  columns = format['sampleCols']
+
+  return {
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/api3.py:289]: Curious: Shouldn't we send it as a `JsonResponse`?

**Review by Harshg999 — COMMENTED** [review]
Suggestion:
Instead of `etl`, would be it better to just use `transformation` in method names, api names, commit message etc. as this only caters for transformation part?

Also maybe good to add comments and docstring wherever possible?

**Pull request closed without merging**